grails-app/services/com/mini/asaas/payment/PaymentService.groovy

```diff
@@ -55,6 +62,26 @@ class PaymentService {
         payment.save(failOnError: true)
     }
 
+    public void processOverdue() {
+        List<Payment> paymentList = paymentService.listByStatus(PaymentStatus.WAITING)
+        
+        paymentList.each { payment ->
+            paymentService.markOverdueIfDue(payment)
+        }
+    }
+
+    private void markOverdueIfDue(Payment payment) {
+        if (verifyIfOverdue(payment)) {
+            payment.status = PaymentStatus.OVERDUE;
+            payment.save();
+        }
+    }
+
+    public Boolean verifyIfOverdue(Payment payment) {        
```

Comment:
Para um método que retorna um Boolean, pode ser usado o is.    is Overdue, neste caso.